### PR TITLE
Cookbook Recipe: SQuAD Extraction

### PR DESCRIPTION
**Pull request by willbakst**
* Using a modified subset of the SQuAD 2.0 dataset to demonstrate how to extract an answer to a question from context.
* Uses Oxen to easily discover what types of mistakes the LLM is making to iterate on and improve the prompts.

### CODE REVIEW

File: cookbook/extraction_examples/squad_extraction/push_to_oxen.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    repo.add(filename)
+    repo.commit(
+        f"Add prompt version {prompt_version} extracted answers joined with questions."
+    )
+    repo.push()
+
+
+def main(prompt_version: str):
+    """Uploads the joined geology-squad data and versioned answers."""
+    config_user("William Bakst", "william@mirascope.io")
```
Comment (brenkao) [inline cookbook/extraction_examples/squad_extraction/push_to_oxen.py:75]: for cookbook example, do you want to use variable NAME and EMAIL instead, or made up name. I see this often in other libraries

File: docs/cookbook/squad_extraction.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+## Initial Basic Extraction
+
+To extract an answer to a question, we can use the [`OpenAIChat.extract`](../api/chat/models/openai_chat.md/#mirascope.chat.models.OpenAIChat.extract) method to extract the answer. First, let's create a super basic schema and prompt for asking the question and extracting an answer:
+
+```shell
+$ mirascope init; touch prompts/question.py
+```
+
+```python
+# prompts/question.py
```
Comment (brenkao) [inline docs/cookbook/squad_extraction.md:32]: in our cookbook example, you used squad_prompts, should we match them?

File: docs/cookbook/squad_extraction.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+from mirascope import Prompt
+from pydantic import BaseModel, Field
+
+
+class ExtractedAnswer(BaseModel):
+    """The anwer to a question about a paragraph of text."""
```
Comment (brenkao) [inline docs/cookbook/squad_extraction.md:38]: typo anwer -> answer

File: docs/cookbook/squad_extraction.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+```
+{
+  "exact": 79.3103448275862,
+  "f1": 89.4777077966733,
+  "total": 116,
+}
+```
+
+Exact match is determined with a raw string match, whereas F1 score is calculated by token to account for partial matches. With our simple script, we were able to get nearly 80% exact match and an F1 score of nearly 90 on our 116 examples. Before we move on, let's include this metadata in the docstring of our prompt file and version it with the CLI:
```
Comment (brenkao) [inline docs/cookbook/squad_extraction.md:104]: 90 -> missing % sign?

**Review by brenkao — APPROVED** [review]

File: cookbook/extraction_examples/squad_extraction/push_to_oxen.py (hunk 1 above)
Comment (willbakst) [reply cookbook/extraction_examples/squad_extraction/push_to_oxen.py:75]: Good catch. I do this in the setup file and forgot to remove this line here.

File: docs/cookbook/squad_extraction.md (hunk 2 above)
Comment (willbakst) [reply docs/cookbook/squad_extraction.md:32]: Good catch.

File: docs/cookbook/squad_extraction.md (hunk 3 above)
Comment (willbakst) [reply docs/cookbook/squad_extraction.md:38]: Done.

File: docs/cookbook/squad_extraction.md (hunk 4 above)
Comment (willbakst) [reply docs/cookbook/squad_extraction.md:104]: This is F1 score, so no percent (vs. exact match which is a percentage).

**Pull request merged**